app/controllers/testOnly/TransitAccompanyingDocumentP5Controller.scala

```diff
@@ -16,39 +16,62 @@
 
 package controllers.testOnly
 
+import cats.data.Validated
+import config.PhaseConfig
 import controllers.actions.AuthenticateActionProvider
+import models.P5.Phase.PostTransition
+import models.P5.Phase.Transition
 import play.api.Logging
 import play.api.mvc.Action
 import play.api.mvc.AnyContent
 import play.api.mvc.ControllerComponents
 import services.P5.DepartureMessageP5Service
+import services.conversion.TransitAccompanyingDocumentConversionService
 import services.pdf.TADPdfGenerator
 import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
 import utils.FileNameSanitizer
 
 import javax.inject.Inject
 import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
 
 class TransitAccompanyingDocumentP5Controller @Inject() (
   pdf: TADPdfGenerator,
   service: DepartureMessageP5Service,
   authenticate: AuthenticateActionProvider,
-  cc: ControllerComponents
-)(implicit ec: ExecutionContext)
+  cc: ControllerComponents,
+  conversionService: TransitAccompanyingDocumentConversionService
+)(implicit ec: ExecutionContext, phaseConfig: PhaseConfig)
     extends BackendController(cc)
     with Logging {
 
   def get(departureId: String, messageId: String): Action[AnyContent] = authenticate().async {
     implicit request =>
-      service.getReleaseForTransitNotification(departureId, messageId).map {
-        ie029 =>
-          val fileName = s"TAD_${FileNameSanitizer(ie029.data.TransitOperation.MRN)}.pdf"
+      service.getIE015MessageId(departureId).flatMap {
+        case Some(ie015MessageId) =>
+          for {
+            ie015 <- service.getDeclarationData(departureId, ie015MessageId)
+            ie029 <- service.getReleaseForTransitNotification(departureId, messageId)
+            genPdf <- phaseConfig.phase match {
+              case PostTransition => Future.successful(Validated.Valid(pdf.generateP5TADPostTransition(ie029)))
+              case Transition     => conversionService.fromP5ToViewModel(ie029, ie015).map(_.map(pdf.generateP5TADTransition))
+            }
+          } yield genPdf match {
```

Comment:
I think it may also be good to keep the logic out of the yield {}, maybe pull that out into a separate function as well: